lib/database/player-game.php

```diff
@@ -275,19 +275,24 @@ function getUsersCompletedGamesAndMax($user): array
     $requiredFlags = AchievementType::OfficialCore;
     $minAchievementsForCompletion = 5;
 
-    $query = "SELECT gd.ID AS GameID, c.Name AS ConsoleName, c.ID AS ConsoleID, gd.ImageIcon, gd.Title, COUNT(ach.GameID) AS NumAwarded, inner1.MaxPossible, (COUNT(ach.GameID)/inner1.MaxPossible) AS PctWon, aw.HardcoreMode
+    $query = "SELECT gd.ID AS GameID, c.Name AS ConsoleName, c.ID AS ConsoleID, gd.ImageIcon, gd.Title, inner1.MaxPossible,
+            ANY_VALUE(aw.HardcoreMode), SUM(aw.HardcoreMode = 0) AS NumAwarded, SUM(aw.HardcoreMode = 1) AS NumAwardedHC,
+            (SUM(aw.HardcoreMode = 0) / inner1.MaxPossible) AS PctWon,
+            (SUM(aw.HardcoreMode = 1) / inner1.MaxPossible) AS PctWonHC
         FROM Awarded AS aw
         LEFT JOIN Achievements AS ach ON ach.ID = aw.AchievementID
         LEFT JOIN GameData AS gd ON gd.ID = ach.GameID
         LEFT JOIN
             ( SELECT COUNT(*) AS MaxPossible, ach1.GameID FROM Achievements AS ach1 WHERE Flags = $requiredFlags GROUP BY GameID )
             AS inner1 ON inner1.GameID = ach.GameID AND inner1.MaxPossible > $minAchievementsForCompletion
         LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-        WHERE aw.User='$user' AND ach.Flags = $requiredFlags
-        GROUP BY ach.GameID, aw.HardcoreMode, gd.Title
-        ORDER BY PctWon DESC, inner1.MaxPossible DESC, gd.Title ";
+        WHERE aw.User = '$user' AND ach.Flags = $requiredFlags
+        GROUP BY ach.GameID, gd.Title
+        ORDER BY (SUM(aw.HardcoreMode = 0) / inner1.MaxPossible) = 1.0 DESC,
+            PctWonHC DESC, PctWon DESC, inner1.MaxPossible DESC, gd.Title";
 
     $db = getMysqliConnection();
+    mysqli_query($db, "SET SESSION innodb_strict_mode=ON");
```

Comment:
Is this required for this change to work as intended?

The whole legacy mysqli database layer will be replaced in the near future.

Changing the innodb session behaviour in mid-flight seems risky to me. I might be missing an important detail here though. If absolutely required please add a comment line or to to explain why it is required in this particular case.

Reply:
Ah no, sorry... That was just a leftover from my previous tests.